src/notation/internal/notationinteraction.cpp

```diff
@@ -3989,6 +3989,8 @@ void NotationInteraction::addIntervalToSelectedNotes(int interval)
     }
 
     if (notes.empty()) {
+        MScore::setError(MsError::NO_NOTE_SELECTED);
+        checkAndShowMScoreError();
```

Comment:
really not a fan of using a static variable for this at all...I know that's how it's been done elsewhere but there's a very good argument for moving away from it where possible (e.g. if MuseScore or any other application wanted to use the engraving library to operate on multiple scores etc.)

Reply:
I'm currently very new to MuseScore development so how it's been done elsewhere is really as much as I can know how to do lol. It did certainly strike me as a strange way to go about it. Is there a specific change I could make to the PR or is revamping this error system in the scope of another PR?

Reply:
I think this is fine for now. Yes, the whole MScore class is an ugly thing with its static variables etc, but this is not something we should saddle first time contributors with. I expect that we will overhaul the code for interactions with the score in the next months, and proper error handling will be one of the important focusses.